pkg/controller/planexecution/planexecution_controller.go

```diff
@@ -238,25 +240,37 @@ func (r *ReconcilePlanExecution) Reconcile(request reconcile.Request) (reconcile
 	}
 
 	// Before returning from this function, update the status
-	defer r.Update(context.Background(), planExecution)
+	defer r.Update(ctx, planExecution)
 
 	// Need to add ownerReference as the Instance.
-	instance.Status.ActivePlan = corev1.ObjectReference{
-		Name:       planExecution.Name,
-		Kind:       planExecution.Kind,
-		Namespace:  planExecution.Namespace,
-		APIVersion: planExecution.APIVersion,
-		UID:        planExecution.UID,
+	if instance.Status.ActivePlan.Name == "" {
```

Comment:
🤔 is this the main fix? is this fix to the issue we were looking at? If so I don't understand it because it feels like this piece of code is not creating PlanExecution CRD...

Reply:
ooh I see, it's a different bug!

Reply:
Updating the `Instance.Status` from the `PlanExecutionController` feels very weird.

I think a better approach would be to have the `InstanceController` watch `PlanExecution`s and calculate + write the `Instance.Status` only in `InstanceController.Reconcile`.

To calculate the status, `InstanceController` would fetch all plan executions owned by the instance, and then assign `instance.Status.ActivePlan.Name` to the name of the latest PlanExecution that is not in a "terminal state".

Reply:
This has been removed... weirdness likely is moved elsewhere :) 

BTW... The writing of PlanExecution was all over the place... it is in PlanExecutationController, InstanceController etc.   
@gkleiman lets have a review of the latest... I think you have some good points here.